storage/src/cache/cachedfile.rs

```diff
@@ -1095,10 +1133,38 @@ impl FileCacheEntry {
             return Ok(0);
         }
         if region.chunks.len() > 1 {
+            let mut meta: Option<Arc<BlobCompressionContextInfo>> = None;
+            // Validate the chunk order.
             for idx in 0..region.chunks.len() - 1 {
-                let end = region.chunks[idx].compressed_offset()
-                    + region.chunks[idx].compressed_size() as u64;
-                let start = region.chunks[idx + 1].compressed_offset();
+                let pre_chunk = &region.chunks[idx];
+                let next_chunk = &region.chunks[idx + 1];
+
+                // Lazy load blob meta info if needed.
+                if (pre_chunk.is_batch() || next_chunk.is_batch()) && meta.is_none() {
```

Comment:
There are repeated code patterns here which makes implementation a little complex. It would be great to use a helper function here such as:
```
struct BlobCCI {
         meta: Option<Arc<BlobCompressionContextInfo>>,
}

impl BlobCCI {
      fn get_chunk_compressed_offset(chunk: &ChunkInfo) -> Result<u64) {
            xxxxx
      }
}
```

Reply:
The helper `BlobCCI` is implemented.